test/basic.js

```diff
@@ -0,0 +1,49 @@
+'use-strict'
+
+const helper = require('./helper.js')
+const { test } = require('tap')
+
+test('Connect-Publish-Disconnect 1000 clients', async function (t) {
+  const aedes = helper.initAedes()
+  const server = await helper.createServer(aedes.handle)
+  const total = 1000
+  var connected = 0
+  var disconnected = 0
+  var published = 0
+
+  aedes.on('client', function (c) {
+    connected++
+  })
+
+  aedes.on('clientDisconnect', function (c) {
+    disconnected++
+  })
+
+  aedes.on('clientError', function (c, error) {
+    t.error('Client error')
+  })
+
+  aedes.on('publish', function (packet, client) {
+    // ignore $SYS topics
+    if (packet.topic === 'my/topic') {
+      published++
+    }
+  })
```

Comment:
I would avoid to create aedes and test the public API. I would literally start it as a black box _in a separate process_.

Reply:
Oh, ok. So should I create a file `broker.js` where I init aedes and with tls, ws and net support support and then create a test suites with clients that connect to that?

Reply:
I have made some changes. Let me know if that is the way, then I need to know how you would like to handle aedes process, should I kill it after all tests are ended?